### PR TITLE
Use pre-built nvcomp 2.3 binaries by default

### PR DESCRIPTION
**Pull request by robertmaynard**
When building cudf on x86_64 hardware we can enable the new pre-built nvcomp 2.3 binaries to leverage new compression codec support.

Closes https://github.com/rapidsai/cudf/issues/10681
Closes https://github.com/rapidsai/cudf/pull/10837

### CODE REVIEW

**Comment by jbrennan333** [comment]
@robertmaynard do you want to pull the changes from #10837 into this PR?

**Comment by robertmaynard** [comment]
> @robertmaynard do you want to pull the changes from #10837 into this PR?

Yeah I will do so

File: build.sh
Comment (jlowe) [inline build.sh:38]:
```suggestion
   --opensource_nvcomp           - disable use of proprietary nvcomp extensions
```

**Review by jlowe — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

File: fetch_rapids.cmake
Comment (jbrennan333) [inline fetch_rapids.cmake:17]: Is this a temporary repo path?

Comment (robertmaynard) [reply fetch_rapids.cmake:17]: Yes, waiting on https://github.com/rapidsai/rapids-cmake/pull/190 to be merged.

Comment (robertmaynard) [reply fetch_rapids.cmake]: rapids-cmake PR has been merged, so this has been removed.

**Comment by jakirkham** [comment]
rerun tests

**Comment by jakirkham** [comment]
Actually think we need PR ( https://github.com/rapidsai/integration/pull/473 ) first to fix CI issues

**Comment by jbrennan333** [comment]
I ran into a problem with the spark-rapids-jni build with these changes.
```
[INFO] --- maven-antrun-plugin:3.0.0:run (build-sparkrapidsjni) @ spark-rapids-jni ---
[INFO] Executing tasks
[INFO]     [mkdir] Created dir: /rapids/spark-rapids-jni/target/cmake-build
...
[INFO]      [exec] -- Configuring incomplete, errors occurred!
[INFO]      [exec] See also "/rapids/spark-rapids-jni/target/cmake-build/CMakeFiles/CMakeOutput.log".
[INFO]      [exec] See also "/rapids/spark-rapids-jni/target/cmake-build/CMakeFiles/CMakeError.log".
[INFO]      [exec] CMake Error at /rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/nvcomp/nvcomp-targets.cmake:101 (message):
[INFO]      [exec]   The imported target "nvcomp::nvcomp" references the file
[INFO]      [exec] 
[INFO]      [exec]      "/rapids/spark-rapids-jni/target/libcudf-install/lib/libnvcomp.so"
[INFO]      [exec] 
[INFO]      [exec]   but this file does not exist.  Possible reasons include:
[INFO]      [exec] 
[INFO]      [exec]   * The file was deleted, renamed, or moved to another location.
[INFO]      [exec] 
[INFO]      [exec]   * An install or uninstall procedure did not complete successfully.
[INFO]      [exec] 
[INFO]      [exec]   * The installation package was faulty and contained
[INFO]      [exec] 
[INFO]      [exec]      "/rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/nvcomp/nvcomp-targets.cmake"
[INFO]      [exec] 
[INFO]      [exec]   but not all the files it references.
[INFO]      [exec] 
[INFO]      [exec] Call Stack (most recent call first):
[INFO]      [exec]   /rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/nvcomp/nvcomp-config.cmake:62 (include)
[INFO]      [exec]   /usr/local/cmake-3.22.3-linux-x86_64/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
[INFO]      [exec]   /rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:25 (find_dependency)
[INFO]      [exec]   /rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
[INFO]      [exec]   /rapids/spark-rapids-jni/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:89 (find_package)
[INFO]      [exec]   CMakeLists.txt:98 (rapids_find_package)

```
It looks like the path may be wrong. It is looking for: `/rapids/spark-rapids-jni/target/libcudf-install/lib/libnvcomp.so` but this is where I see it: `/rapids/spark-rapids-jni/target/libcudf-install/lib64/libnvcomp.so`

**Comment by robertmaynard** [comment]
> I ran into a problem with the spark-rapids-jni build with these changes.
> 
> ```
> [INFO] --- maven-antrun-plugin:3.0.0:run (build-sparkrapidsjni) @ spark-rapids-jni ---
> [INFO] Executing tasks
> [INFO]     [mkdir] Created dir: /rapids/spark-rapids-jni/target/cmake-build
> ...
> [INFO]      [exec] -- Configuring incomplete, errors occurred!
> [INFO]      [exec] See also "/rapids/spark-rapids-jni/target/cmake-build/CMakeFiles/CMakeOutput.log".
> [INFO]      [exec] See also "/rapids/spark-rapids-jni/target/cmake-build/CMakeFiles/CMakeError.log".
> [INFO]      [exec] CMake Error at /rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/nvcomp/nvcomp-targets.cmake:101 (message):
> [INFO]      [exec]   The imported target "nvcomp::nvcomp" references the file
> [INFO]      [exec] 
> [INFO]      [exec]      "/rapids/spark-rapids-jni/target/libcudf-install/lib/libnvcomp.so"
> [INFO]      [exec] 
> [INFO]      [exec]   but this file does not exist.  Possible reasons include:
> [INFO]      [exec] 
> [INFO]      [exec]   * The file was deleted, renamed, or moved to another location.
> [INFO]      [exec] 
> [INFO]      [exec]   * An install or uninstall procedure did not complete successfully.
> [INFO]      [exec] 
> [INFO]      [exec]   * The installation package was faulty and contained
> [INFO]      [exec] 
> [INFO]      [exec]      "/rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/nvcomp/nvcomp-targets.cmake"
> [INFO]      [exec] 
> [INFO]      [exec]   but not all the files it references.
> [INFO]      [exec] 
> [INFO]      [exec] Call Stack (most recent call first):
> [INFO]      [exec]   /rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/nvcomp/nvcomp-config.cmake:62 (include)
> [INFO]      [exec]   /usr/local/cmake-3.22.3-linux-x86_64/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
> [INFO]      [exec]   /rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/cudf/cudf-dependencies.cmake:25 (find_dependency)
> [INFO]      [exec]   /rapids/spark-rapids-jni/target/libcudf-install/lib64/cmake/cudf/cudf-config.cmake:91 (include)
> [INFO]      [exec]   /rapids/spark-rapids-jni/target/cmake-build/_deps/rapids-cmake-src/rapids-cmake/find/package.cmake:89 (find_package)
> [INFO]      [exec]   CMakeLists.txt:98 (rapids_find_package)
> ```
> 
> It looks like the path may be wrong. It is looking for: `/rapids/spark-rapids-jni/target/libcudf-install/lib/libnvcomp.so` but this is where I see it: `/rapids/spark-rapids-jni/target/libcudf-install/lib64/libnvcomp.so`

https://github.com/rapidsai/rapids-cmake/pull/194 should fix this issue

**Comment by robertmaynard** [comment]
> I believe @robertmaynard is out. Was someone else going to pick this up? Not sure how to address the ARM build issues.

I am out, put I pushed a change to this PR that hopefully will fix the error we are seeing on ARM builds. Issue was due to a bad filename.

**Comment by jbrennan333** [comment]
I verified that the spark-rapids-jni issue is fixed.  Thanks @robertmaynard!

**Review by jbrennan333 — APPROVED** [review]
+1 this looks good to me.

File: conda/recipes/libcudf/post-link.sh
Comment (gerashegalov) [inline conda/recipes/libcudf/post-link.sh:5]: nit: spaces around `>>`

File: build.sh
Comment (vyasr) [inline build.sh:149]: I find it a bit disconcerting that all of the flags used for the JNI build do not respect the script's CLI. That seems like it's just waiting to bite some unsuspecting user of `build.sh` eventually. It looks like this isn't a new problem though since it affects all of the flags used in the JNI build, though, so we don't need to block this PR over this. It's probably worth factoring this build logic into a separate script thought since AFAICT it completely ignores all of the logic and variables in this build.sh script.

Comment (vyasr) [inline build.sh:149]:
```suggestion
                -DCUDF_USE_PROPRIETARY_NVCOMP=ON \
```

File: conda/recipes/libcudf/post-link.sh
Comment (vyasr) [inline conda/recipes/libcudf/post-link.sh:5]:
```suggestion
  cat ./nvlink.txt >> $PREFIX/.messages.txt
```

File: fetch_rapids.cmake
Comment (vyasr) [inline fetch_rapids.cmake:14]:
```suggestion
```

**Review by vyasr — CHANGES_REQUESTED** [review]

File: conda/recipes/libcudf/post-link.sh
Comment (vyasr) [reply conda/recipes/libcudf/post-link.sh]: I had the same comment in my review and I pushed the fix.

**Comment by jbrennan333** [comment]
@vyasr I was able to build spark-rapids-jni.
@jlowe brought up a question off-line - I don't see anything in the code here that specifies which version of nvcomp we are downloading?   Is it fixed to nvcomp-2.3?   We don't really want it just pulling down the latest version, do we?
That said, we have found a problem in nvcomp-2.3 zstd codec, so there will likely be an updated version that we will need (2.3.1?).

**Comment by jbrennan333** [comment]
To answer my own question, the version we download is specified here: https://github.com/rapidsai/rapids-cmake/blob/branch-22.06/rapids-cmake/cpm/versions.json

**Comment by vyasr** [comment]
Yep, sorry for the slow response. Versions for packages that we need to be standardized across all of rapids are stored in the rapids-cmake file that you pointed to. The rapids-cmake function is being called in the get_nvcomp.cmake file in this repository.

**Comment by jbrennan333** [comment]
Closes https://github.com/rapidsai/cudf/issues/10681

**Comment by vyasr** [comment]
Just to give an update for those tracking this PR, we're holding off on merge while we wait to resolve issues with nvcomp 2.3.0

**Review by jbrennan333 — APPROVED** [review]

**Review by vyasr — APPROVED** [review]
@robertmaynard based on our meeting I think we are good to move forward with this, correct? I'll approve and let you merge when ready.

**Comment by robertmaynard** [comment]
@gpucibot merge

**Comment by robertmaynard** [comment]
@gpucibot merge

**Pull request merged**
